frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -287,22 +284,48 @@ function ReaderFooter:updateFooterContainer()
     if self.settings.align == "left" then
         self.footer_container = LeftContainer:new{
             dimen = Geom:new{ w = 0, h = self.height },
-            self.footer_content,
+            self.horizontal_group
         }
     elseif self.settings.align == "right" then
         self.footer_container = RightContainer:new{
             dimen = Geom:new{ w = 0, h = self.height },
-            self.footer_content,
+            self.horizontal_group
         }
     else
         self.footer_container = CenterContainer:new{
             dimen = Geom:new{ w = 0, h = self.height },
-            self.footer_content,
+            self.horizontal_group
+        }
+    end
+    local separator_line = LineWidget:new{
+        dimen = Geom:new{
+            w = Screen:getWidth() - 2 * self.horizontal_margin,
+            h = Size.line.medium,
         }
+    }
+    local vertical_span = VerticalSpan:new{width = self.bottom_padding *2}
+
+    if not self.settings.bottom_horizontal_separator then
+        vertical_span.width = 0
+        separator_line.dimen.h = 0
     end
+    self.vertical_frame = VerticalGroup:new{
+        separator_line,
+        vertical_span,
+        self.footer_container,
+    }
```

Comment:
I'd rather not see `separator_line` and `vertical_span` when `not self.settings.bottom_horizontal_separator `, instead of giving them a height of 0 (because that's not pretty :) and may be one day, they won't support a height of 0 :)
So, rather:
```lua
if self.settings.bottom_horizontal_separator then
  local separator_line = ...
  local vertical_span = ...
  self.vertical_frame = VerticalGroup:new{
        separator_line,
        vertical_span,
        self.footer_container,
    }
else
  self.vertical_frame = VerticalGroup:new{
        self.footer_container,
    }
end
```
It looks like we always access things by name, nowhere with indices like `self.vertical_frame[3]`, so that should work.
(Even just `self.vertical_frame = self.footer_container` might work, but may be less risky keeping the same hierarchy.)

Reply:
And even if we did use `self.vertical_frame[3]` that would generally make it a piece of code that should be changed to a named reference for clarity and maintainability instead.